grails-app/controllers/com/unifina/controller/api/ProductApiController.groovy

```diff
@@ -130,6 +130,8 @@ class ProductApiController {
 	def uploadImage(String id) {
 		Product product = productService.findById(id, loggedInUser(), Permission.Operation.WRITE)
 		MultipartFile file = getUploadedFile()
+		// TODO(kkn): Replace file.bytes with file.getInputStream()
```

Comment:
I've been playing around with an idea that instead of copying byte arrays around just pass the InputStream instead. But it seems that image scaling will copy the byte array anyway.